### PR TITLE
View function

### PR DESCRIPTION
**Pull request by tayruxin**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_VIEW_COMPANY_SUCCESS = "Viewing Company: %1$s";
+
+    private final Index targetIndex;
+
+    /**
+     * Creates an ViewCommand to view the specified {@code Company}
+     * @param targetIndex index of the company in the filtered company list to view
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:33]: I think there should be a space between description and @params. Missing full stops too.

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        return new CommandResult(String.format(MESSAGE_VIEW_COMPANY_SUCCESS, companyDetailsToDisplayString));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:63]: Do you mean null cases instead of nulls?

**Review by GlendaChong — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.company.Company;
+
+/**
+ * Deletes a company identified using it's displayed index from the address book.
```
[the remark this answers is not in the document]
Comment (tayruxin) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java]: Resolved

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            + ": Views the company identified by the index number used in the displayed company list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_VIEW_COMPANY_SUCCESS = "Viewing Company: %1$s";
+
+    private final Index targetIndex;
+
+    /**
+     * Creates an ViewCommand to view the specified {@code Company}.
```
Comment (GlendaChong) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:32]: Creates "a" ViewCommand...

File: src/main/java/seedu/address/logic/commands/ViewCommand.java (hunk 1 above)
Comment (GlendaChong) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java:34]: According to the [Javadocs standards](https://se-education.org/guides/conventions/java/intermediate.html#comments), there should be punctuations behind each parameter description.

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Parses input arguments and creates a new DeleteCommand object
+ */
+public class ViewCommandParser implements Parser<ViewCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ViewCommand
+     * and returns a ViewCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
[the remark this answers is not in the document]
Comment (GlendaChong) [reply src/main/java/seedu/address/logic/parser/ViewCommandParser.java:18]: Likewise, missing full stop here too

**Review by GlendaChong — APPROVED** [review]
Generally LGTM, though there's still some minor styling issues to be fixed

**Pull request merged**